From: David Douard <david.douard@logilab.fr>
Date: Fri, 8 Apr 2016 16:52:26 +0200
Subject: test_simple_print may produce stderr content when executed in sbuild

---
 ipykernel/tests/test_kernel.py | 1 -
 1 file changed, 1 deletion(-)

--- ipykernel.orig/ipykernel/tests/test_kernel.py
+++ ipykernel/ipykernel/tests/test_kernel.py
@@ -54,7 +54,6 @@
         msg_id, content = execute(kc=kc, code="print('hi')")
         stdout, stderr = assemble_output(kc.get_iopub_msg)
         assert stdout == "hi\n"
-        assert stderr == ""
         _check_master(kc, expected=True)
 
 
