From: Julien Puydt <julien.puydt@laposte.net>
Date: Sun, 2 Oct 2016 14:32:58 +0200
Subject: Made build reproducible -- patch by Daniel Shahaf

---
 ipykernel/kernelspec.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipykernel/kernelspec.py b/ipykernel/kernelspec.py
index c0cc9ff..2cce2cb 100644
--- a/ipykernel/kernelspec.py
+++ b/ipykernel/kernelspec.py
@@ -100,7 +100,7 @@ def write_kernel_spec(
     if overrides:
         kernel_dict.update(overrides)
     with open(pjoin(path, "kernel.json"), "w") as f:
-        json.dump(kernel_dict, f, indent=1)
+        json.dump(kernel_dict, f, indent=1, sort_keys=True)
 
     return str(path)
 
