From: Julien Puydt <julien.puydt@laposte.net>
Date: Sun, 2 Oct 2016 14:32:58 +0200
Subject: Made build reproducible -- patch by Daniel Shahaf

---
 ipykernel/kernelspec.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- ipykernel.orig/ipykernel/kernelspec.py
+++ ipykernel/ipykernel/kernelspec.py
@@ -87,7 +87,7 @@
     if overrides:
         kernel_dict.update(overrides)
     with open(pjoin(path, "kernel.json"), "w") as f:
-        json.dump(kernel_dict, f, indent=1)
+        json.dump(kernel_dict, f, indent=1, sort_keys=True)
 
     return path
 
